editors/emacs/Portfile

```diff
@@ -142,9 +143,20 @@ if {$subport eq "emacs-devel" || $subport eq "emacs-app-devel"} {
 
     livecheck.type none
 
+    # Emacs 30 removes libjansson in favour of native JSON support.
+    configure.args-delete --with-json
+    depends_lib-delete port:jansson
+    configure.args-append --with-native-compilation=no
+
+    # TODO: Put this somewhere better. See https://github.com/emacs-mirror/emacs/commit/4c6f45fa8eef1a15d5790c1f3d3e608b548015db#diff-731c852b7c27c6ee90c44c871e2988ecfc2d225bc720156d7a47c1d563c2683aR59
```

Comment:
what is the purpose of this comments; what would be a better place to put it? And if there is such a place why not do it now?

Reply:
Ah yes. Do you want this as a new variant option or always present by default? The linked comment:

```
** New configuration option '--disable-gc-mark-trace'.
This disables the GC mark trace buffer for about 5 % better garbage
collection performance.  Doing so may make it more difficult for Emacs
developers to help finding GC-related bugs that you run into, which is
why it the mark trace buffer is enabled by default.
```

Reply:
well, it depends - do you think many users will be "Emacs developers" as that seems to be the usual situation where you might not want to have it disabled. I would probably have it use this configure.arg always; if @drkp has a different opinion he can always change it and/or put it in a variant.

Reply:
agree - I think we should leave this disabled by default, and if someone really wants it enabled for development we can create a variant. (Or if they're emacs developers, they can probably figure it out themselves. :-)

Reply:
I will leave it as-is and not create a variant then.